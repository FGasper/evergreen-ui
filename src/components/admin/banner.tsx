import * as React from "react";
import { Card, CardContent, CardHeader, TextField } from '@material-ui/core';

interface Props {
    onBannerTextChange?:(text:string) => void;
    banner?:string;
}

export class BannerCard extends React.Component<Props> {
    public render() {
        return (
            <Card>
                <CardHeader title="Announcements" />
                <CardContent>
                    <TextField label="Banner Message" value={this.props.banner || ""} onChange={this.setBanner()}/>
                </CardContent>
            </Card>
        )
    }

    private setBanner = () => (event: React.ChangeEvent<HTMLInputElement>) => {
        if (this.props.onBannerTextChange) {
            this.props.onBannerTextChange(event.currentTarget.value);
        }
      }
}