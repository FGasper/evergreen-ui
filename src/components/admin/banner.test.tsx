import * as React from 'react';
import * as enzyme from 'enzyme';
import { BannerCard } from './banner';

test("BannerCard renders", () => {
    const onChange = (s:string) => console.log("onChange called");
    const wrapper = enzyme.render(<BannerCard banner="hello world" onBannerTextChange={onChange} />);
    expect(wrapper.html()).toContain("hello world");
    expect(wrapper.html()).toContain("Banner Message");
})