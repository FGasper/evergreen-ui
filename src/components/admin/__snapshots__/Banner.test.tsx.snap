// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerCard matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BannerCard
    banner="hello world"
    onBannerTextChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(CardHeader)
          title="Announcements"
        />,
        <WithStyles(CardContent)>
          <TextField
            id="bannerText"
            label="Banner Message"
            onChange={[Function]}
            required={false}
            select={false}
            value="hello world"
            variant="standard"
          />
        </WithStyles(CardContent)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": "Announcements",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TextField
            id="bannerText"
            label="Banner Message"
            onChange={[Function]}
            required={false}
            select={false}
            value="hello world"
            variant="standard"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "bannerText",
            "label": "Banner Message",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "hello world",
            "variant": "standard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(CardHeader)
            title="Announcements"
          />,
          <WithStyles(CardContent)>
            <TextField
              id="bannerText"
              label="Banner Message"
              onChange={[Function]}
              required={false}
              select={false}
              value="hello world"
              variant="standard"
            />
          </WithStyles(CardContent)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": "Announcements",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TextField
              id="bannerText"
              label="Banner Message"
              onChange={[Function]}
              required={false}
              select={false}
              value="hello world"
              variant="standard"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "bannerText",
              "label": "Banner Message",
              "onChange": [Function],
              "required": false,
              "select": false,
              "value": "hello world",
              "variant": "standard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
